packages/flutter/lib/src/painting/text_editing.dart

```diff
@@ -0,0 +1,137 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Whether a [TextPosition] is visually upstream or downstream of its offset.
+///
+/// When a text position exists at a line break, a single offset has two visual
```

Comment:
This should be clearer that the line break thing is only an example. I had misunderstood this example as being the only use case at first.